### PR TITLE
Display taxon breadcrumbs for Education A/B test

### PR DESCRIPTION
**Pull request by alecgibson**
Show taxon breadcrumbs when content is tagged to a taxon, and the user is in the correct A/B testing bucket.

https://trello.com/c/6TLMKxot/337-changes-to-education-related-content-pages-as-part-of-the-new-navigation-in-smartanswers

### CODE REVIEW

File: app/controllers/smart_answers_controller.rb [hunk 1]
```diff
@@ -123,4 +126,13 @@ def setup_navigation_helpers_and_content_item
...
   end
+
+  def breadcrumbs
+    return {} if @navigation_helpers.nil?
+    if should_present_new_navigation_view?
+      @navigation_helpers.taxon_breadcrumbs
+    else
+      @navigation_helpers.breadcrumbs
+    end
+  end
```
Comment (alecgibson) [inline app/controllers/smart_answers_controller.rb]: (See discussion here: https://github.com/alphagov/frontend/pull/1109#discussion_r98249455)

**Comment by tijmenb** [comment]
Requests that are A/B tested need to return a `Vary` header so Fastly knows to vary on our AB header.

But I'd rather have that this is fixed by using the client library, so that the test helpers can detect this problem. Marking DNM until we've started that.

**Comment by alecgibson** [comment]
@tijmenb this is now using the officially published A/B Testing gem.

File: test/functional/smart_answers_controller_test.rb [hunk 2]
```diff
@@ -193,6 +194,54 @@ def teardown
         assert_select "pre.debug", false, "The page should not render debug information"
       end
     end
+
+    context "A/B testing" do
+      setup do
+        # Switch on feature flag
+        ENV['ENABLE_NEW_NAVIGATION'] = 'yes'
```
Comment (mobaig) [inline test/functional/smart_answers_controller_test.rb]: Unless ENV is being reset somewhere this is leaking state. Probably best to stub access to that key.

Comment (alecgibson) [inline test/functional/smart_answers_controller_test.rb]: Yup, about to push a teardown

**Comment by alecgibson** [comment]
@carvil I've also bumped the versions of `govuk_navigation_helpers` and `govuk_ab_testing` on this. Also deployed to int, and can confirm that breadcrumbs work as expected.

**Comment by carvil** [comment]
@alecgibson thanks, could you please add the beta label as well?

**Comment by alecgibson** [comment]
@carvil done

**Pull request merged**